graphql_compiler/schema_generation/sqlalchemy/scalar_type_mapper.py

```diff
@@ -175,15 +176,26 @@
 )
 
 
-def try_get_graphql_scalar_type(column_name, column_type):
+def try_get_graphql_scalar_type(
+    column_name: str, column_type: TypeEngine
+) -> Optional[GraphQLScalarType]:
     """Return the matching GraphQLScalarType for the SQL datatype or None if none is found."""
-    maybe_graphql_type = SQL_CLASS_TO_GRAPHQL_TYPE.get(type(column_type), None)
-    if maybe_graphql_type is None:
-        # Trying to get the string representation of the SQLAlchemy JSON and ARRAY types
-        # will lead to an error. We therefore use repr instead.
+    # HACK(pmantica1): The code above implies that the GraphQL type is just a factor of the class
+    #                  of column_type. However, it is also a function of the attributes of
+    #                  column_type. We should probably refactor this file to better reflect that.
```

Comment:
I don't see the problem with the code, or why a refactor is necessary.

Reply:
I was thinking then this could be hacky in the future once we add a timezone aware GraphQL datetime type. Then it wouldn't be possible to map SQL types to zero-or-one GraphQL types. So I don't think that a dictionary would be the right construct. However, this code is not really hacky right now so I'll remove the comment. (Also, I don't know if I explained myself above, but I don't think the issue I described is relevant yet, so let's not discuss it any further).